iib/workers/tasks/build_containerized_fbc_operations.py

```diff
@@ -0,0 +1,330 @@
+    attributes=get_binary_versions(),
+)
+def handle_containerized_fbc_operation_request(
+    request_id: int,
+    fbc_fragments: List[str],
+    from_index: str,
+    binary_image: Optional[str] = None,
+    distribution_scope: str = '',
+    overwrite_from_index: bool = False,
+    overwrite_from_index_token: Optional[str] = None,
+    build_tags: Optional[List[str]] = None,
+    add_arches: Optional[Set[str]] = None,
+    binary_image_config: Optional[Dict[str, Dict[str, str]]] = None,
+    index_to_gitlab_push_map: Dict[str, str] = {},
+    used_fbc_fragment: bool = False,
+) -> None:
+    """
+    Add fbc fragments to an fbc index image.
+
+    :param list fbc_fragments: list of fbc fragments that need to be added to final FBC index image
+    :param int request_id: the ID of the IIB build request
+    :param str binary_image: the pull specification of the container image where the opm binary
+        gets copied from.
+    :param str from_index: the pull specification of the container image containing the index that
+        the index image build will be based from.
+    :param set add_arches: the set of arches to build in addition to the arches ``from_index`` is
+        currently built for; if ``from_index`` is ``None``, then this is used as the list of arches
+        to build the index image for
+    :param dict index_to_gitlab_push_map: the dict mapping index images (keys) to GitLab repos
+        (values) in order to push their catalogs into GitLab.
+    :param bool used_fbc_fragment: flag indicating if the original request used fbc_fragment
+        (single) instead of fbc_fragments (array). Used for backward compatibility.
+    """
+    _cleanup()
+    set_request_state(request_id, 'in_progress', 'Resolving the fbc fragments')
+
+    # Resolve all fbc fragments
+    resolved_fbc_fragments = []
+    for fbc_fragment in fbc_fragments:
+        with set_registry_token(overwrite_from_index_token, fbc_fragment, append=True):
+            resolved_fbc_fragment = get_resolved_image(fbc_fragment)
+            resolved_fbc_fragments.append(resolved_fbc_fragment)
+
+    prebuild_info = prepare_request_for_build(
+        request_id,
+        RequestConfigFBCOperation(
+            _binary_image=binary_image,
+            from_index=from_index,
+            overwrite_from_index_token=overwrite_from_index_token,
+            add_arches=add_arches,
+            fbc_fragments=fbc_fragments,
+            distribution_scope=distribution_scope,
+            binary_image_config=binary_image_config,
+        ),
+    )
+
+    from_index_resolved = prebuild_info['from_index_resolved']
+    binary_image_resolved = prebuild_info['binary_image_resolved']
+    arches = prebuild_info['arches']
+    # Variable mr_details needs to be assigned, otherwise
+    # cleanup_on_failure() fails when an exception is raised.
+    mr_details: Optional[Dict[str, str]] = None
+
+    with set_registry_token(overwrite_from_index_token, fbc_fragment, append=True):
```

Comment:
Do we want to add this here now? Or should we add it to `set_opm_version` like we discussed?